crates/shared/src/oneinch_api.rs

```diff
@@ -554,15 +554,24 @@ where
     D: DeserializeOwned,
 {
     tracing::trace!(%url, "Query 1inch API");
-    let response = client.get(url).send().await?.text().await?;
-    tracing::trace!(%response, "Received 1Inch API response");
-
-    match serde_json::from_str::<RestResponse<D>>(&response)? {
-        RestResponse::Ok(result) => Ok(result),
-        RestResponse::Err(err) => {
+    let response = client.get(url).send().await?;
+    let status_code = response.status();
+    let response = response.text().await?;
+    tracing::trace!(%response, ?status_code, "Received 1Inch API response");
```

Comment:
Do we handle `status_code == 429` in the caller?

Reply:
We currently handle it when converting it to a `TradeError`. Not sure if there are other places we need to check this case now.

Reply:
Since the `OneInchPriceEstimator` is built on top of `dyn TraitFinding` this should be enough as the conversion from `TradeError` to `PriceEstimationError` is already `Err(RateLimited)` aware:
```rust
impl From<OneInchError> for TradeError {
    fn from(err: OneInchError) -> Self {
        match err {
            OneInchError::Api(err) if err.status_code == 429 => Self::RateLimited,
            err if err.is_insuffucient_liquidity() => Self::NoLiquidity,
            err => Self::Other(err.into()),
        }
    }
}
```

Reply:
Looks like we don't propagate a `RateLimited` variant from the 1Inch error, so we need to add that as well.

Reply:
Might make sense for `OneInchError` to have a `RateLimited` error variant.

Reply:
Will merge like this for now and clean up when there is no time pressure.